lib/ansible/cli/galaxy.py

```diff
@@ -199,59 +297,129 @@ def _display_role_info(role_info):
+
     def execute_init(self):
         """
-        creates the skeleton framework of a role that complies with the galaxy metadata format.
+        Creates the skeleton framework of a role or collection that complies with the Galaxy metadata format.
         """
 
+        galaxy_type = context.CLIARGS['type']
         init_path = context.CLIARGS['init_path']
         force = context.CLIARGS['force']
-        role_skeleton = context.CLIARGS['role_skeleton']
+        obj_skeleton = context.CLIARGS['{0}_skeleton'.format(galaxy_type)]
 
-        role_name = context.CLIARGS['role_name']
-        role_path = os.path.join(init_path, role_name)
-        if os.path.exists(role_path):
-            if os.path.isfile(role_path):
-                raise AnsibleError("- the path %s already exists, but is a file - aborting" % role_path)
-            elif not force:
-                raise AnsibleError("- the directory %s already exists."
-                                   "you can use --force to re-initialize this directory,\n"
-                                   "however it will reset any main.yml files that may have\n"
-                                   "been modified there already." % role_path)
+        obj_name = context.CLIARGS['{0}_name'.format(galaxy_type)]
 
         inject_data = dict(
-            role_name=role_name,
             author='your name',
             description='your description',
             company='your company (optional)',
             license='license (GPL-2.0-or-later, MIT, etc)',
             issue_tracker_url='http://example.com/issue/tracker',
-            min_ansible_version='2.4',
-            role_type=context.CLIARGS['role_type']
+            repository_url='http://example.com/repository',
+            documentation_url='http://docs.example.com',
+            homepage_url='http://example.com',
+            min_ansible_version=ansible_version[:3],  # x.y
+            ansible_plugin_list_dir=get_versioned_doclink('plugins/plugins.html'),
         )
 
-        # create role directory
-        if not os.path.exists(role_path):
-            os.makedirs(role_path)
+        if galaxy_type == 'role':
+            inject_data['role_name'] = obj_name
+            inject_data['role_type'] = context.CLIARGS['role_type']
+            obj_path = os.path.join(init_path, obj_name)
+        elif galaxy_type == 'collection':
+            namespace, collection_name = obj_name.split('.', 1)
 
-        if role_skeleton is not None:
+            inject_data['namespace'] = namespace
+            inject_data['collection_name'] = collection_name
+            obj_path = os.path.join(init_path, namespace, collection_name)
+
+        if os.path.exists(obj_path):
+            if os.path.isfile(obj_path):
+                raise AnsibleError("- the path %s already exists, but is a file - aborting" % obj_path)
+            elif not force:
+                raise AnsibleError("- the directory %s already exists."
```

Comment:
A whitespace or `\n` is missing:
```suggestion
                raise AnsibleError("- the directory %s already exists. "
```
Current version produces ` exists.You `